processor.go

```diff
@@ -331,6 +332,57 @@ func (p *CertProcessor) secretName(cert Certificate) string {
 	return p.certSecretPrefix + cert.Spec.Domain
 }
 
+// normalizeHostnames returns a copy of the hostnames array where all hostnames are lower
+// cased and the array sorted.
+// This allows the input to have changed order or different casing between runs,
+// but a new certificate will only be created if a certificate is added or removed.
+func normalizeHostnames(hostnames []string) []string {
+	arr := make([]string, len(hostnames))
+	copy(arr, hostnames)
+	for i, hostname := range arr {
+		arr[i] = strings.ToLower(hostname)
+	}
+	sort.Strings(arr)
+
+	return arr
+}
+
+func (p *CertProcessor) getStoredAltNames(cert Certificate) ([]string, error) {
+	var altNamesRaw []byte
+	err := p.db.View(func(tx *bolt.Tx) error {
+		altNamesRaw = tx.Bucket([]byte("domain-altnames")).Get([]byte(cert.Spec.Domain))
+		return nil
+	})
+	if err != nil {
+		return nil, errors.Wrapf(err, "Error while fetching altnames from database for domain %v", cert.Spec.Domain)
+	}
+	if altNamesRaw == nil {
+		altNamesRaw, err = json.Marshal([]string{})
```

Comment:
it seems like it would be easier to `return []string{}, nil` here rather than go to the trouble of marshal+unmarshal to do the same.